build.sbt

```diff
@@ -40,7 +40,7 @@ lazy val standardSettings = Defaults.defaultSettings ++ Seq(
     "-language:higherKinds",
     "-language:implicitConversions",
     "-unchecked",
-    "-Xfatal-warnings",
+    //"-Xfatal-warnings", temporarly removed in order for the whole thing to build. If we could change this setting to be selectively applied in the build script, then it might make sense to remove it permanently here, anyway...
```

Comment:
Ah, why do we need to delete the old api and tests? Can we just add the new stuff alongside the old and then delete the old when we're done?